src/UnitTests/ForEachEnumeratorDispose.cs

```diff
@@ -0,0 +1,343 @@
+﻿namespace AutoMapper.UnitTests
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using Shouldly;
+    using Xunit;
+
+    public class ForEachEnumeratorDispose : NonValidatingSpecBase
```

Comment:
The same tests need to work for struct enumerators. Check the `List` source code for an example.

Reply:
Why NonValidatingSpecBase? The configuration should be valid.

Reply:
Also there are no tests for the error case.